### ISSUE TITLE
Avoid to delete 'build' packages

### ISSUE THREAD

**Issue by pgreze** [rock3r/deep-clean#1]
About https://github.com/rock3r/deep-clean/blob/e6b9acbd8ed6a8ff975c9ff75239460b812680b2/deep-clean.kts#L64
some border cases could happen, like:

- subprojects/core/src/main/java/org/gradle/build/XX.java
- .env/whatever/build

The safest solution would be to use "gradle clean" but some rules like ignoring:

- .../src/.../build
- build folders in hidden files

Could be an easy way to fix them.

Nice script by the way!

**Comment by rock3r** [comment]
Thanks! I have in mind to add a `--backup` option that won't actually delete stuff but just rename it so you can change things back if you have such issues. I may consider adding exclusions too, at some point, but I don't really have an immediate need so I'm not sure when it'll happen.

**Comment by rock3r** [comment]
The backup option has been added in [1.2.0](https://github.com/rock3r/deep-clean/releases/tag/1.2.0)

**Comment by rock3r** [comment]
I actually realised that the original script wasn't recursively traversing folders, so this couldn't have happened. In [1.3.0](https://github.com/rock3r/deep-clean/releases/tag/1.3.0) I have fixed the issue and added a `--not-recursive` option to prevent accidental recursions. I'd still recommend using `--backup` and maybe `--dry-run` before actually running the script "for real", just in case.

I'm not sure whether I should consider this done; the `--not-recursive` option should save you from most headaches, but also greatly dials down the effectiveness of this script. Maybe we'll have a file-based excludes list at some point, but I don't really want to do it right now.
